ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/config_manager.py

```diff
@@ -0,0 +1,106 @@
+
+    It should be initialized per device and the instance should be passed to the AvdTestBase classe.
+    """
+
+    def __init__(self, device_name: str, hostvars: Mapping) -> None:
+        """Initialize the ConfigManager class.
+
+        Args:
+        ----
+            device_name (str): The current device name for which the plugin is being run.
+            hostvars (Mapping): A mapping that contains a key for each device with a value of the structured_config.
+                                      When using Ansible, this is the `task_vars['hostvars']` object.
+        """
+        self.device_name = device_name
+        self.hostvars = hostvars
+        self.structured_config = self.get_host_structured_config(host=device_name)
+        self.loopback0_mapping = self.get_loopback0_mapping()
+        self.vtep_mapping = self.get_vtep_mapping()
+
+    def get_host_structured_config(self, host: str) -> Mapping:
+        """Get a specified host's structured configuration from the hostvars.
+
+        Args:
+        ----
+            host (str): Hostname to retrieve the structured_config.
+
+        Returns:
+        -------
+            dict: Structured configuration for the host.
+
+        Raises:
+        ------
+            AristaAvdError: If host is not in hostvars or if its structured_config is not a mapping object.
+        """
+        if host not in self.hostvars:
+            raise AristaAvdError(message=f"Host '{host}' is missing from the hostvars.")
+        struct_cfg = self.hostvars[host]
+
+        # Check if struct_cfg is a mapping object (e.g. Ansible 'hostvars' object or regular dict)
+        if not isinstance(struct_cfg, Mapping):
+            raise AristaAvdError(message=f"Host '{host}' structured_config is not a dictionary or dictionary-like object.")
+
+        return struct_cfg
+
+    def get_loopback0_mapping(self) -> list[tuple[str, str]]:
+        """Get the loopback0_mapping list."""
+        return self._get_loopback_mappings["loopback0_mapping"]
+
+    def get_vtep_mapping(self) -> list[tuple[str, str]]:
+        """Get the vtep_mapping list."""
+        return self._get_loopback_mappings["vtep_mapping"]
+
+    @cached_property
+    def _get_loopback_mappings(self) -> dict:
+        """Generate the loopback mappings for the eos_validate_state tests, which are used in AvdTestBase subclasses.
+
+        Returns
+        -------
+            dict: A dictionary containing:
+            - "loopback0_mapping": A list of tuples where each tuple contains a hostname and its Loopback0 IP address.
+            - "vtep_mapping": A list of tuples where each tuple contains a hostname and its VTEP IP address if `Vxlan1` is the source_interface.
+
+        """
+        results = {"loopback0_mapping": [], "vtep_mapping": []}
+
+        for host in self.hostvars:
+            host_struct_cfg = self.get_host_structured_config(host)
```

Comment:
Yes it makes things hard since the plugin is run per device. I don't think you can share stuff super easily unless we create/use something similar to `EosDesignsFacts`.